datumaro/plugins/synthetic_data/utils.py

```diff
@@ -0,0 +1,163 @@
+# Copyright (C) 2022 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+from pathlib import Path
+import os.path as osp
+
+import cv2 as cv
+import numpy as np
+import requests
+
+
+class IFSFunction:
+    def __init__(self, rng, prev_x, prev_y):
+        self.function = []
+        self.xs, self.ys = [prev_x], [prev_y]
+        self.select_function = []
+        self.cum_proba = 0.0
+        self._rng = rng
+
+    def set_param(self, params, proba, weights=None):
+        if weights is not None:
+            params = list(np.array(params) * np.array(weights))
+
+        self.function.append(params)
+        self.cum_proba += proba
+        self.select_function.append(self.cum_proba)
+
+    def calculate(self, iterations):
+        rand = [self._rng.random() for _ in range(iterations)]
+        prev_x, prev_y = 0, 0
+        next_x, next_y = 0, 0
+
+        for i in range(iterations):
+            for func_params, select_func in zip(self.function, self.select_function):
+                a, b, c, d, e, f = func_params
+                if rand[i] <= select_func:
+                    next_x = prev_x * a + prev_y * b + e
+                    next_y = prev_x * c + prev_y * d + f
+                    break
+
+            self.xs.append(next_x)
+            self.ys.append(next_y)
+            prev_x = next_x
+            prev_y = next_y
+
+    @staticmethod
+    def process_nans(data):
+        nan_index = np.where(np.isnan(data))
+        extend = np.array(range(nan_index[0][0] - 100, nan_index[0][0]))
```

Comment:
Where does the number 100 come from?

Reply:
From original repo: https://github.com/hirokatsukataoka16/FractalDB-Pretrained-ResNet-PyTorch/blob/d25b5ad641c45b5ec7cd94d74378233af563e8e1/fractal_renderer/ifs_function.py#L79

Reply:
So what does it mean?

Reply:
I couldn't find any information in paper about this number